test/helper.lua

```diff
@@ -192,4 +192,49 @@ function helpers.assert_ge(actual, expected, message)
     end
 end
 
+function helpers.get_replicasets_with_equal_key(cluster, key)
```

Comment:
Seems that this function can be replced by `get_other_storage_bucket_id` function from `custom_bucket_id_test`.

Moreover, `get_other_storage_bucket_id` name looks better.
This function does get replica sets, but in fact, only one of them corresponds to the specified key.
The main idea of this function is to get bucket_id that corresponds to other replicas than the specified key (in fact, bucket_id).

I think that this function can be common for your test and `custom_bucket_id_test`.
The signature can be changed to `get_other_storage_bucket_id(cluster, bucket_id)` since you need to use both bucket IDs.